python/ray/tune/tests/test_searchers.py

```diff
@@ -0,0 +1,155 @@
+            num_samples=100,
+            reuse_actors=True)
+
+        best_trial = out.best_trial
+        self.assertLessEqual(best_trial.config["report"], 2.0)
+
+    def testDragonfly(self):
+        from ray.tune.suggest.dragonfly import DragonflySearch
+
+        out = tune.run(
+            _invalid_objective,
+            search_alg=DragonflySearch(domain="euclidean", optimizer="random"),
+            config=self.config,
+            metric="_metric",
+            mode="max",
+            num_samples=100,
+            reuse_actors=True)
+
+        best_trial = out.best_trial
+        self.assertLessEqual(best_trial.config["point"], 2.0)
+
+    def testHyperopt(self):
+        from ray.tune.suggest.hyperopt import HyperOptSearch
+
+        out = tune.run(
+            _invalid_objective,
+            search_alg=HyperOptSearch(),
+            config=self.config,
+            metric="_metric",
+            mode="max",
+            num_samples=100,
+            reuse_actors=True)
+
+        best_trial = out.best_trial
+        self.assertLessEqual(best_trial.config["report"], 2.0)
+
+    def testNevergrad(self):
+        from ray.tune.suggest.nevergrad import NevergradSearch
+        import nevergrad as ng
+
+        out = tune.run(
+            _invalid_objective,
+            search_alg=NevergradSearch(optimizer=ng.optimizers.OnePlusOne),
+            config=self.config,
+            metric="_metric",
+            mode="max",
+            num_samples=100,
+            reuse_actors=True)
+
+        best_trial = out.best_trial
+        self.assertLessEqual(best_trial.config["report"], 2.0)
+
+    def testOptuna(self):
+        from ray.tune.suggest.optuna import OptunaSearch
+
+        out = tune.run(
+            _invalid_objective,
+            search_alg=OptunaSearch(),
+            config=self.config,
+            metric="_metric",
+            mode="max",
+            num_samples=100,
+            reuse_actors=True)
+
+        best_trial = out.best_trial
+        self.assertLessEqual(best_trial.config["report"], 2.0)
+
+    def testSkopt(self):
+        from ray.tune.suggest.skopt import SkOptSearch
+
+        out = tune.run(
+            _invalid_objective,
+            search_alg=SkOptSearch(),
+            config=self.config,
+            metric="_metric",
+            mode="max",
+            num_samples=100,
+            reuse_actors=True)
+
+        best_trial = out.best_trial
+        self.assertLessEqual(best_trial.config["report"], 2.0)
+
+    def testZOOpt(self):
+        from ray.tune.suggest.zoopt import ZOOptSearch
+        np.random.seed(1234)
+
+        out = tune.run(
+            _invalid_objective,
+            search_alg=ZOOptSearch(budget=8, parallel_num=8),
+            config=self.config,
+            metric="_metric",
+            mode="max",
+            num_samples=8,
+            reuse_actors=True)
+
+        best_trial = out.best_trial
+        self.assertLessEqual(best_trial.config["report"], 2.0)
```

Comment:
does it make sense to abstract this into a helper function?

Reply:
There's some special cases for some of the algorithms for seeding in the last changes, thus I'd like to keep it as is for now